packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart

```diff
@@ -31,50 +20,60 @@ mixin AuthenticatorPhoneField<FieldType,
         ? inputResolver.resolve(context, widget.hintTextKey!)
         : widget.hintText!;
 
-    return Row(
-      children: [
-        DropdownButton<String>(
-          value: selectionValue,
-          items: countryCodeStrings.map((int value) {
-            return DropdownMenuItem<String>(
-              value: value.toString(),
-              child: Text('+$value'),
-            );
-          }).toList(),
-          onChanged: (newValue) {
-            setState(() {
-              selectionValue = newValue!;
-            });
-          },
-        ),
-        Expanded(
-          child: TextFormField(
-            style: enabled
-                ? null
-                : const TextStyle(
-                    color: AmplifyColors.black20,
-                  ),
-            initialValue: initialValue,
-            enabled: enabled,
-            validator: validator,
-            onChanged: (phoneValue) {
-              onChanged.call('+$selectionValue$phoneValue');
-            },
-            decoration: InputDecoration(
-              suffixIcon: suffixIcon,
-              errorMaxLines: errorMaxLines,
-              focusedBorder: OutlineInputBorder(
-                borderSide: BorderSide(
-                  color: Theme.of(context).primaryColor,
+    final countryResolver = stringResolver.countries;
+    return Focus(
+
+        /// Display the prefix only when phone number widget is in focus
+        onFocusChange: (hasFocus) {
+          setState(() {
+            _prefixWidth = hasFocus ? 40 : 0;
+          });
+        },
+        child: TextFormField(
+          style: enabled
+              ? null
+              : const TextStyle(
+                  color: AmplifyColors.black20,
                 ),
+          initialValue: initialValue,
+          enabled: enabled,
+          validator: widget.validatorOverride ?? validator,
+          onChanged: onChanged,
+          decoration: InputDecoration(
+            prefix: SizedBox(
+                height: 15,
+                width: _prefixWidth,
+                child: PopupMenuButton(
+                  initialValue: selectionValue,
+                  child: Center(child: Text('+$selectionValue')),
+                  itemBuilder: (context) {
+                    return countryCodes.map((Country country) {
+                      return PopupMenuItem<String>(
+                        value: country.value,
+                        child: Text(
+                            '${countryResolver.resolve(context, country.key)} (+${country.value})',
```

Comment:
Probably out of scope, but I think some customers may want to localize `value` as well. I assume cognito requires 0-9, but some other languages use other numerals and customers may want to display those.

I would say this is out of scope because we probably need more info on requirements around this if it is something we want to support it. Adding some sort of concept of "value" (what is sent to cognito) versus a localized "displayValue" (what displays on the screen) would probably meet these use cases, but I am not totally sure.

Reply:
Yeah, maybe we can just display `${countryResolver.resolve(context, country.key)}`, and then our defauly en_arb can include the '+$value'?